### PR TITLE
Feature/data table

### PR DESCRIPTION
**Pull request by xkilpoi**
### Description

New features including
- column based filtering
- HTML templates for columns
- HTML templates for child rows
- persistence of pagination / sort / filter

### Which Component is affected or generated?

data-table.module

### CODE REVIEW

File: projects/material-addons/src/lib/data-table/data-table.component.spec.ts [hunk 1]
```diff
@@ -472,21 +490,24 @@ describe('DataTableComponent', () => {
       const initialData: any[] = [
         { rowId: 1, id: 1, name: 'Zebra' },
         { rowId: 2, id: 2, name: 'Alligator' },
+        //TODO why is it that way?
```
Comment (freesbi) [inline projects/material-addons/src/lib/data-table/data-table.component.spec.ts:493]: You're right. I missed to add "id" field to 3rd row

File: projects/material-addons/src/lib/data-table/data-table.component.spec.ts [hunk 2]
```diff
@@ -472,21 +490,24 @@ describe('DataTableComponent', () => {
...
+      const getAllDataSourceRowsOfCurrentPageSpy = jest.spyOn(component, 'filteredPageData', 'get');
       component.onToggleSelectAll();
 
       expect(clearSpy).toHaveBeenCalled();
       expect(component.allSelected).toBeTruthy();
       expect(getAllDataSourceRowsOfCurrentPageSpy).toHaveBeenCalledTimes(1);
       expect(getAllDataSourceRowsOfCurrentPageSpy).toHaveReturnedWith(component.dataSource.data);
-      expect(selectSpy).toHaveBeenNthCalledWith(1, '1');
-      expect(selectSpy).toHaveBeenNthCalledWith(2, '2');
+      //TODO I guess this was an error in the test and mine is fine now? - why strings before?
```
Comment (freesbi) [inline projects/material-addons/src/lib/data-table/data-table.component.spec.ts:507]: Now it looks better. Thanks for this fix

File: projects/material-addons/src/lib/data-table/data-table-filter/data-table-filter.module.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { NgModule } from "@angular/core";
+import { CommonModule } from "@angular/common";
+import { DataTableFilter } from "./data-table-filter.directive";
+import { DataTableFilterComponentModule } from "./data-table-filter-component/data-table-filter.component.module";
+import { DataTableFilterHeader } from "./data-table-filter-header.directive";
+
+@NgModule({
```
Comment (Wagei) [inline projects/material-addons/src/lib/data-table/data-table-filter/data-table-filter.module.ts:7]: is there a reason to use modules? we migrated the whole project to standalone 1 year ago. Can you please change all new components/directives to standalone?

**Review by Wagei — APPROVED** [review]
It looks good to me, please take a look to my comment! It is only a small change.

**Pull request merged**